### DISCUSSION TITLE
Invoking a Template on a Site More Than Once

### DISCUSSION THREAD

**Discussion by anthonywhite** [pnp/pnpframework#335] [Q&A] [answered]
Is there any documentation/guidance on intended behaviours when re-applying a template to a site? E.g. a new version of a template delivering a mixture of new and updated artifacts?

It appears that e.g. existing List Views will get updated with new fields, but List View web parts on existing client side pages do not - so pages would have to be deleted before re-running the template.

**Comment by jansenbe** [comment, marked as answer]
@anthonywhite : classic list web parts indeed do take a snapshot of the view when being put on a page, changing the view afterwards will not update the view used by the web part. Overwriting the pages (or first deleting them) should help
